client/src/connection/rest/util.ts

```diff
@@ -38,23 +38,48 @@ export const getResourceIdFromItem = (item: ContentItem): string | null => {
   return getLink(item.links, "GET", "self")?.uri || null;
 };
 
-export const getSasContentUri = (item: ContentItem, readOnly?: boolean): Uri =>
-  Uri.parse(
-    `${readOnly ? `${ContentSourceType.SASContent}ReadOnly` : ContentSourceType.SASContent}:/${
-      item.name
-        ? item.name.replace(/#/g, "%23").replace(/\?/g, "%3F")
-        : item.name
-    }?id=${getResourceIdFromItem(item)}`,
+export const getSasContentUri = (
+  item: ContentItem,
+  readOnly?: boolean,
+  fullPath?: string,
+): Uri => {
+  let pathToUse = fullPath || item.name || "";
+
+  // Remove leading slash if present since we'll add one in the URI construction
+  if (pathToUse.startsWith("/")) {
+    pathToUse = pathToUse.substring(1);
+  }
+
+  // URL encode special characters
+  pathToUse = pathToUse.replace(/#/g, "%23").replace(/\?/g, "%3F");
+
+  return Uri.parse(
+    `${readOnly ? `${ContentSourceType.SASContent}ReadOnly` : ContentSourceType.SASContent}:/${pathToUse}?id=${getResourceIdFromItem(item)}`,
   );
+};
+
+export const getSasServerUri = (
+  item: ContentItem,
+  readOnly?: boolean,
+  fullPath?: string,
+): Uri => {
+  let pathToUse = fullPath || item.name || "";
 
-export const getSasServerUri = (item: ContentItem, readOnly?: boolean): Uri =>
-  Uri.parse(
-    `${readOnly ? `${ContentSourceType.SASServer}ReadOnly` : ContentSourceType.SASServer}:/${
-      item.name
-        ? item.name.replace(/#/g, "%23").replace(/\?/g, "%3F")
-        : item.name
-    }?id=${getResourceIdFromItem(item)}`,
+  // Decode SAS server path encoding (~fs~ represents /)
+  pathToUse = pathToUse.replace(/~fs~/g, "/");
+
+  // Remove leading slash if present since we'll add one in the URI construction
+  if (pathToUse.startsWith("/")) {
+    pathToUse = pathToUse.substring(1);
+  }
+
+  // URL encode special characters
+  pathToUse = pathToUse.replace(/#/g, "%23").replace(/\?/g, "%3F");
```

Comment:
Could you move this into a shared function and call it from `getSasContentUri` and `getSasServerUri`. You could pass in a "replacements" regex to handle the difference in paths
```
pathToUse = replacement ? pathToUse.replace(replacement, "/") : pathToUse;
```